src/Compilers/CSharp/Portable/Symbols/FunctionPointers/FunctionPointerTypeSymbol.cs

```diff
@@ -157,5 +157,22 @@ internal override TypeSymbol SetNullabilityForReferenceTypes(Func<TypeWithAnnota
             return this;
         }
 
+        internal static bool RefKindEquals(TypeCompareKind compareKind, RefKind refKind1, RefKind refKind2)
+        {
+            if ((compareKind & TypeCompareKind.FunctionPointerRefMatchesOutInRefReadonly) != 0)
+            {
+                return (refKind1, refKind2) switch
```

Comment:
>(refKind1, refKind2) switch [](start = 23, length = 27)

Why so complicated? It looks like MemberSignatureComparer simply does  ```(refKind1 == RefKind.None) == (refKind2 == RefKind.None)```

Reply:
I found this significantly more readable than the version in `MemberSignatureComparer`.

---
In reply to: [426089324](https://github.com/dotnet/roslyn/pull/44257#discussion_r426089324) [](ancestors = 426089324)

Reply:
> I found this significantly more readable than the version in MemberSignatureComparer.

I spent significant amount of time trying to figure out your intent here (time I could have spent elsewhere) and whether all the right combinations are handled and handled properly. Every reader will have to go through the same trouble over and over again. I understand your desire to use switch expressions everywhere you can, but in this case, I believe it makes code less clear and unnecessarily complicated. Please use simplified form of the check, especially that we are already using it elsewhere. 

---
In reply to: [426821892](https://github.com/dotnet/roslyn/pull/44257#discussion_r426821892) [](ancestors = 426821892,426089324)

Reply:
>  I understand your desire to use switch expressions everywhere you can

To be clear, that is not the motivation.

> I spent significant amount of time trying to figure out your intent here (time I could have spent elsewhere) and whether all the right combinations are handled and handled properly. 

And _I_ spent probably a similar amount of time creating a truth table in my head for the original check to be sure that I could understand what it was checking. I used this form because _I_ genuinely find it significantly easier to read. I don't find the other from simpler in any fashion.

---
In reply to: [426825848](https://github.com/dotnet/roslyn/pull/44257#discussion_r426825848) [](ancestors = 426825848,426821892,426089324)

Reply:
I'll change this. However, I want to be clear that I don't just use new features for the fun of it. I use them when it would help me to read the code better, and I find that writing out the truth table (as I can do concisely here with a switch expression) is significantly easier to understand. I'll change this because we have prior art in the form of a comparison, but in future code that does similar things I'm still going to prefer using the form I understand more easily.

---
In reply to: [426840351](https://github.com/dotnet/roslyn/pull/44257#discussion_r426840351) [](ancestors = 426840351,426825848,426821892,426089324)